packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -396,7 +399,8 @@ export const transformPlotlyJsonToSankeyProps = (
     }))
     // eslint-disable-next-line @typescript-eslint/no-shadow
     //@ts-expect-error Dynamic link object. Ignore for now.
-    .filter(x => x.source !== x.target); // Filter out self-references (circular links)
+    // Filter out non-negative nodes, unequal nodes and self-references (circular links)
+    .filter(x => x.source > 0 && x.target > 0 && x.source !== x.target);
```

Comment:
circular links can also happen if source1 -> target1 and target1 -> source1. Are we also handling this scenario?